pkg/nuctl/command/patch.go

```diff
@@ -0,0 +1,366 @@
+		outputManifest: map[string]interface{}{
+			"Failed":  map[string]error{},
+			"Success": []string{},
+		},
+		outputManifestLock: sync.Mutex{},
+	}
+
+	cmd := &cobra.Command{
+		Use:     "functions",
+		Aliases: []string{"func", "fn", "function"},
+		Short:   "patch functions",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// initialize root
+			if err := patchCommandeer.rootCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize root")
+			}
+
+			// initialize patch commandeer
+			if err := patchCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize patch commandeer")
+			}
+
+			return commandeer.patchFunctions(ctx)
+		},
+	}
+
+	cmd.PersistentFlags().StringSliceVarP(&commandeer.excludedProjects, "exclude-projects", "", []string{}, "Exclude projects to patch")
+	cmd.PersistentFlags().StringSliceVarP(&commandeer.excludedFunctions, "exclude-functions", "", []string{}, "Exclude functions to patch")
+	cmd.PersistentFlags().IntVarP(&commandeer.concurrency, "concurrency", "c", 10, "Max number of parallel patches")
+	cmd.PersistentFlags().StringVarP(&commandeer.username, "username", "u", "", "Username of a user with permissions to patch functions")
+	cmd.PersistentFlags().StringVarP(&commandeer.password, "password", "p", "", "Password of a user with permissions to patch functions")
+	cmd.PersistentFlags().StringVarP(&commandeer.apiURL, "api-url", "", "", "URL of the nuclio API (e.g. https://nuclio.io:8070)")
+	cmd.PersistentFlags().BoolVarP(&commandeer.waitForFunction, "wait", "w", false, "Wait for function to be ready after patching")
+
+	// mark required flags
+	cmd.MarkPersistentFlagRequired("username") // nolint: errcheck
+	cmd.MarkPersistentFlagRequired("password") // nolint: errcheck
+	cmd.MarkPersistentFlagRequired("base-url") // nolint: errcheck
+
+	commandeer.cmd = cmd
+
+	return commandeer
+}
+
+func (c *patchFunctionsCommandeer) patchFunctions(ctx context.Context) error {
+
+	functionNames, err := c.getFunctionNames(ctx)
+	if err != nil {
+		return errors.Wrap(err, "Failed to get function names")
+	}
+
+	c.rootCommandeer.loggerInstance.DebugWith("Got function names", "functionNames", functionNames)
+
+	// create session
+	sessionCookieHeader, err := c.createSession(ctx)
+	if err != nil {
+		return errors.Wrap(err, "Failed to create session")
+	}
+
+	patchErrGroup, _ := errgroup.WithContextSemaphore(ctx, c.rootCommandeer.loggerInstance, uint(c.concurrency))
+	for _, function := range functionNames {
+		function := function
+		patchErrGroup.Go("patch function", func() error {
+			err := c.patchFunction(ctx, function, sessionCookieHeader)
+			c.addResultToOutputManifest(function, err)
+			if err != nil {
+				return errors.Wrap(err, "Failed to patch function")
+			}
+			return nil
+		})
+	}
+
+	if err := patchErrGroup.Wait(); err != nil {
+		c.rootCommandeer.loggerInstance.WarnWith("Failed to patch functions", "err", err)
```

Comment:
maybe add a comment that the errnous functions are printed/logged on the output manifest and thus you ignore the err  and not failing the entire operation